### PR TITLE
Ratvar will now bitchslap BOH babies

### PR DESCRIPTION
**Pull request by Robustin**
:cl: Robustin
balance: Clockwork magicks will now prevent Bags of Holding from being combined on Reebe. 
/:cl:

Common sense. I should probably remove the free energy for BOH sings but thats another PR ill never get around to.

Impress me and make a PA + Sing Generator next time you want to cheese the ending of Clock Cult.

### CODE REVIEW

**Review by Cyberboss — DISMISSED** [review]

**Comment by Buggy123** [comment]
90% of the time the BoH just eats the attackers anyway, because for some reason the singulo's movement is weirdly not random and tends to move south on Reebee.

**Comment by RogueSteampunker** [comment]
Show me a time when BoH bombing Rebee actually won a round, and then we'll talk.

**Comment by vuonojenmustaturska** [comment]
For what its worth, i've seen it happen twice

**Comment by JJRcop** [comment]
You forgot to set them on fire and delete both bags.

**Comment by ishitbyabullet** [comment]
@RogueSteampunker a singularity allowed the crew to reach the ark. Only 4-10 people, including clock cultists and me, survived to reach the ark and beat it down while avoiding getting beaten down by the surviving cultists.

**Comment by DoctoreMad** [comment]
>Show me a time when BoH bombing Rebee actually won a round

You know I've been part of one. Context: Cultists were mostly distracted by other attackers so the RD threw himself on a crate with 2 Bags of Holding. Form then, we just wait for engineering to damage the start of the defenses, get to the upper part, drill down the airlocks, drop the crate with the bomber, then have RIPLEY Bodyblock the cultists form stopping it.
Worked like a charm.

**Comment by ShizCalev** [comment]
>approved
>doesn't even compile

File: code/game/objects/items/storage/backpack.dm [hunk 1]
```diff
@@ -71,6 +71,12 @@
 
 /obj/item/storage/backpack/holding/handle_item_insertion(obj/item/W, prevent_warning = 0, mob/user)
 	if((istype(W, /obj/item/storage/backpack/holding) || count_by_type(W.GetAllContents(), /obj/item/storage/backpack/holding)))
+		var/turf/loccheck = get_turf(src)
+		if(loccheck.z == ZLEVEL_CITYOFCOGS)
+			 to_chat(user, "<span class='inathneq_large'><i>I don't think so...</i></span>"
```
Comment (ShizCalev) [inline code/game/objects/items/storage/backpack.dm]: bracket at the end.

File: code/game/objects/items/storage/backpack.dm [hunk 2]
```diff
@@ -71,6 +71,12 @@
 
 /obj/item/storage/backpack/holding/handle_item_insertion(obj/item/W, prevent_warning = 0, mob/user)
 	if((istype(W, /obj/item/storage/backpack/holding) || count_by_type(W.GetAllContents(), /obj/item/storage/backpack/holding)))
+		var/turf/loccheck = get_turf(src)
+		if(loccheck.z == ZLEVEL_CITYOFCOGS)
+			 to_chat(user, "<span class='inathneq_large'><i>I don't think so...</i></span>"
+			 user.visible_message("<span class='inathneq_small'>An unseen force knocks [user] to the ground!</span>")
```
Comment (ChangelingRain) [inline code/game/objects/items/storage/backpack.dm]: remove like 76 and change this one to `user.visible_message("<span class='warning'>An unseen force knocks [user] to the ground!</span>", "<span class='big_brass'>\"What do you think you're doing?\"</span>")`
ratvar uses gold spans, not blue, and visible 'a bad thing has happened to someone' messages use the warning span.

**Comment by optimumtact** [comment]
At what point do we say that this feature has been driven so far into the realm of "safety and obscure checks for gamemodes" that it's not worth it and we just cut it out completely?

It's a nice reference, but it apparently doesn't seem to be working at all

**Comment by Supermichael777** [comment]
One we get that portal to the netherworld the discords been kicking down the road we could have it toned down from apocalypse to oh shit you fucked up.

**Comment by ACCount12** [comment]
At what point do we say that this game mode has been driven so far into the realm of immersion breaking checks for players shenanigans and snowflake immunities that it's not worth it and we just cut it out completely?

**Comment by Kromgar** [comment]
@ACCount12 seriously it has so many snowflake bullshit rules like how for some reason their clockwork armor is atmosphere proof and can't be harmed by ash storms

**Comment by Repukan** [comment]
Why the fuck is their armor ash proof

**Comment by ChangelingRain** [comment]
the answer is that it isn't unless ratvar is up
and the ash storm staff is probably broken, too, so...

**Comment by optimumtact** [comment]
@ACCount12 there's a big difference between a game mode and a single item

File: code/game/objects/items/storage/backpack.dm (hunk 2 above)
Comment (Armhulen) [reply code/game/objects/items/storage/backpack.dm]: yes, only ratvar bitch slaps people, so joan's comment is much more canon.

**Comment by KorPhaeron** [comment]
This ruins my immersion, ratvar isnt awake yet to slap people the entire mode is about waking him up

**Comment by Armhulen** [comment]
if he's awake enough to kill blood cultists trying armor he's awake enough to bitch slap some guy

**Comment by imsxz** [comment]
Clockwork armor isnt snowflaked to be ash proof, it's just that anything fireproof is immune to ash.

**Comment by Repukan** [comment]
Not really?

Explorer suits are fireproof but not ash proof.

**Comment by lzimann** [comment]
This is stupid, either you change entirely how the boh bomb works or stop adding a zlevel check so only a specific gamemode(and it's ALWAYS clockcult with the special checks) is safe from it

**Comment by Robustin** [comment]
When a mode needs its own bloody Z level to minimize shenanigans, its fair to add a couple z-level checks to accommodate that setup.

My bomb change simply reduced the explosive radius and it appears to have worked out well, this would be the FIRST hard block for a mechanic on Reebe so its hardly some long string of snowflake PRs that must be stopped. 

However I threw this together quickly so I would consider replacing the block with a weaker mechanic (i.e. the 1x1 anchored sing) but I worry about abuse there too.

**Comment by lzimann** [comment]
The gamemode needing its own zlevel is also plain stupid, not to mention it loading literally every round.

**Comment by KorPhaeron** [comment]
I think it is unreasonable to expect snowflake checks to never exist given the very lax rules and sandbox nature of the game

**Comment by Robustin** [comment]
Agreed, stuff like CM has proven that there is a significant audience for a more focused game mode that offers a great deal of depth in exchange for cutting out some of the breadth of a typical round. 

I don't think that should be the "normal" SS13 experience but it strikes me as needlessly purist to say that we can't have a deviation from the typical round either. I still think CC needs a shitload of work but I also think there's a niche for a defensive antag when done properly - but "properly" means that some of the WMD's we "love" to see on a 200x200 space station don't play out so well in a more condensed 60x30 (im just guessing) arena.

**Comment by RandomMarine** [comment]
Players that want a more focused game mode that offers a great deal of depth should go play fuckin' dota and leave the crazy atmospherics simulator alone.

**Comment by Qbopper** [comment]
^ disagree, there's nothing wrong with a different type of round to break things up

if EVERY round was like clock cult MAYBE i'd agree but it's the same as how not every round is like revs

**Comment by DoctoreMad** [comment]
>it's the same as how not every round is like revs

But all TGstation rounds are shit so that's not very different.

**Comment by Qbopper** [comment]
i mean not every round is a straight deathmatch and that's about it but sick meme bro

**Comment by DaxDupont** [comment]
Singularities usually kill the crew instead of the cultists which is hilarious.
Even then most of the time it's not released close to the ark which results in deadchat with their anuses collectively clenching, praying to the RNG gods.

**Comment by Robustin** [comment]
I kinda enjoy the tension of Singularity v. Ark but as it stands its simply too easy to BOH bomb and its a fucking rule nightmare for both teams.

Unlike bombs where the outcome of your actions are somewhat foreseeable, sings are a fucking wildcard and even taking every reasonable precaution could screw over everyone on your team. A cultist popping the bags right above where the crew are could still have it float due north and eat the cult and the ark. A human who pops it outside the ark as a last resort could still have it float south and eat a couple mechs that just showed up to save the day. 

So while I usually support uncommon mechanics that kill a lot of people - if only for the entertainment value - sings are just a little TOO unpredictable and effective. It's easy to ban someone who exploded bombs nowhere near the Ark because they clearly just wanted to kill a bunch of people and see their bombs in action as opposed to actually helping their team, but with a Sing it can feel wrong to punish people based on RNG.

**Comment by Repukan** [comment]
Of course it was Joan lmao

**Comment by Xhuis** [comment]
Not really a fan of this PR, and it was my own fault for missing it when it was on the tracker. Since it's pretty clear that my own word on the mode means nothing at this point, though, I'll wait and see how it plays before I try to revert it.

**Comment by jughu** [comment]
To be fair could you really blame this pr when techwebs allowed you make boh bombs 5 mins into the roumd

**Pull request merged**